### ISSUE TITLE
MD007: list continuation block quote with a list causes `rumdl` to loose track of required indent level

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#526]
### rumdl version

0.1.47

### Installation method

mise

### Bug description

Rule MD007 (https://github.com/rvben/rumdl/blob/8c1798b13cf961c88a4c5d1136448f4f4391383d/docs/md007.md) ensures consistent indentation on lists.

However, it's possible to insert, as list continuation text, a block quote with its own list and cause `rumdl` to lose track of the required indentation for nested items.

### Steps to reproduce

1. Create a Markdown file with this content:
   ```markdown
   ---
   title: Heading
   ---
   
   1. This is a list item:
   
      > This is a note with a list:
      >
      > - List item.
      > - List item.
   
      - This is a list item.
      - This is a list item.
   ```
1. Run `rumdl` on Markdown file with default settings.

### Expected behavior

Given the scenario above, I'd expect no violations.

### Actual behavior

Given the scenario above, `rumdl` returns the following:

```shell
test.md:12:1: [MD007] Expected 2 spaces for indent depth 1, found 3 [*]
test.md:13:1: [MD007] Expected 2 spaces for indent depth 1, found 3 [*]
```

### Configuration

None.

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in v0.1.48. Blockquoted lists no longer corrupt the MD007 parent list indent stack. Thanks for reporting, @eread!
